MSAL/src/native_auth/public/MSALNativeAuthPublicClientApplication.swift

```diff
@@ -68,6 +71,8 @@ public final class MSALNativeAuthPublicClientApplication: MSALPublicClientApplic
             throw MSALNativeAuthInternalError.invalidAuthority
         }
 
+        MSALNativeAuthPublicClientApplication.sharedConfiguration = config
```

Comment:
Should we set this shared config/challenge types after the super.init? I'm saying this because the super.init can throw an error, and we want to assign this shared config only when we know that a new publicClientApplication instance is created. Same comment applies for the other init